### PR TITLE
add Serializable and Cloneable support via JSON string

### PR DESCRIPTION
**Pull request by nrhope**
In many legacy applications JSONObjects endup being part of larger objects that need to be deep-cloned via Serialization or because they are part of Value Objects included in UI Form bean etc. Hence is is very helpful for this class to support such functionality using the very handy and already present .toString() representation as a JSON string. If it isn't supported by the object itself, then an in-built wrapper utility object would still be better than users having to craft their own wrappers.
If this looks like something you'd be willing to accept into the library then I'd be happy to extend to to any other objects like JSONArray and/or add test cases as per your recommendations. 
Let me know what you think, thanks for a great library!

### CODE REVIEW

**Comment by stleary** [comment]
@nrhope Thanks, but I think this is outside the scope of the application. However, I will leave this PR up for a few days in case anyone wants to comment on it.

**Comment by nrhope** [comment]
No problems.

File: src/main/java/org/json/JSONObject.java [hunk 1]
```diff
@@ -81,7 +83,14 @@
  * @author JSON.org
  * @version 2016-08-15
  */
-public class JSONObject {
+public class JSONObject implements Serializable, Cloneable {
```
Comment (javadev) [inline src/main/java/org/json/JSONObject.java:86]: It may be implemented with custom class.

**Comment by stleary** [comment]
Closing due to lack of activity, and prior decision.

**Pull request closed without merging**